### PR TITLE
wss for websockets on tls and single page application behavior

### PR DESCRIPTION
**Pull request by dvaldivia**
Detects `https` and sets the Websocket protocol to `wss`
on asset not found we return `index.html` so the react application can handle 404

### CODE REVIEW

File: portal-ui/src/screens/Console/Logs/Logs.tsx [hunk 1]
```diff
@@ -79,7 +79,13 @@ const Logs = ({
...
     const port = isDev ? "9090" : url.port;
 
-    const c = new W3CWebSocket(`ws://${url.hostname}:${port}/ws/console`);
+    let wsProtocol = "ws";
+    if (url.protocol == "https:") {
+      wsProtocol = "wss";
+    }
+    const c = new W3CWebSocket(
+      `${wsProtocol}://${url.hostname}:${port}/ws/console`
+    );
```
Comment (Alevsk) [inline portal-ui/src/screens/Console/Logs/Logs.tsx:88]: This is almost duplicated code with `Trace.tsx`, can we put this on the `utils.ts` function?

File: portal-ui/src/screens/Console/Trace/Trace.tsx [hunk 2]
```diff
@@ -61,7 +61,14 @@ const Trace = ({
     const isDev = process.env.NODE_ENV === "development";
     const port = isDev ? "9090" : url.port;
 
-    const c = new W3CWebSocket(`ws://${url.hostname}:${port}/ws/trace`);
+    let wsProtocol = "ws";
+    if (url.protocol == "https:") {
+      wsProtocol = "wss";
+    }
```
Comment (cesnietor) [inline portal-ui/src/screens/Console/Trace/Trace.tsx:67]: we can probably put this in a function if all ws will be using it

File: restapi/configure_mcs.go [hunk 3]
```diff
@@ -167,11 +167,44 @@ func FileServerMiddleware(next http.Handler) http.Handler {
...
 				AssetInfo: portalUI.AssetInfo,
-				Prefix:    "build"}).ServeHTTP(w, r)
+				Prefix:    "build"}
+			//assets.Open("index.html")
+			wrapHandlerSinglePageApplication(http.FileServer(&assets)).ServeHTTP(w, r)
+
 		}
 	})
 }
+
```
Comment (cesnietor) [inline restapi/configure_mcs.go:181]: does it need to be an exported struct? if so, all exported need to have comments on top of it.
```
comment on exported type NotFoundRedirectRespWr should be of the form "NotFoundRedirectRespWr ..." (with optional leading article)
```

File: restapi/configure_mcs.go [hunk 4]
```diff
@@ -167,11 +167,44 @@ func FileServerMiddleware(next http.Handler) http.Handler {
...
+	if status != http.StatusNotFound {
+		w.ResponseWriter.WriteHeader(status)
+	}
+}
+
+func (w *NotFoundRedirectRespWr) Write(p []byte) (int, error) {
+	if w.status != http.StatusNotFound {
+		return w.ResponseWriter.Write(p)
+	}
+	return len(p), nil // Lie that we successfully written it
```
Comment (cesnietor) [inline restapi/configure_mcs.go:198]:
```suggestion
	return len(p), nil // Lie that we successfully wrote it
```

File: restapi/configure_mcs.go [hunk 5]
```diff
@@ -167,11 +167,44 @@ func FileServerMiddleware(next http.Handler) http.Handler {
...
+}
+
```
Comment (cesnietor) [inline restapi/configure_mcs.go:200]: please add comments

File: restapi/configure_mcs.go (hunk 3 above)
Comment (cesnietor) [reply restapi/configure_mcs.go:180]: how is this addressed?

Comment (dvaldivia) [reply restapi/configure_mcs.go:180]: unexported

**Pull request merged**